### PR TITLE
[Runtime] Use dyld to find the image path when possible.

### PR DESCRIPTION
**Pull request by al45tair**
For performance reasons, we want to avoid `dladdr()` when possible.

rdar://130954606

### CODE REVIEW

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Comment by al45tair** [comment]
@swift-ci Please smoke test

File: stdlib/public/runtime/Paths.cpp [hunk 1]
```diff
@@ -524,7 +534,8 @@ _swift_initRuntimePath(void *) {
 #if APPLE_OS_SYSTEM
   const char *path = dyld_image_path_containing_address(_swift_initRuntimePath);
 
-  runtimePath = ::strdup(path);
+  // No need to ::strdup() this, apparently
```
Comment (mikeash) [inline stdlib/public/runtime/Paths.cpp:537]: I feel like someone is going to come by in five years and be very confused about "apparently." Suggest, "as the return value from dyld_image_path_containing_address is guaranteed to remain valid as long as the library is loaded."

**Review by mikeash — APPROVED** [review]

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Pull request merged**